### ISSUE TITLE
Widget: Tapping on a certain date opens app with current day

### ISSUE THREAD

**Issue by duerrhannes** [FossifyOrg/Calendar#233]
### Checklist

- [X] I can reproduce the bug with the latest version given [here](https://github.com/FossifyOrg/Calendar/releases/latest).
- [X] I made sure that there are **no existing issues** - [open](https://github.com/FossifyOrg/Calendar/issues) or [closed](https://github.com/FossifyOrg/Calendar/issues?q=is%3Aissue+is%3Aclosed) - to which I could contribute my information.
- [X] I made sure that there are **no existing discussions** - [open](https://github.com/FossifyOrg/Calendar/discussions) or [closed](https://github.com/FossifyOrg/Calendar/discussions?discussions_q=is%3Aclosed) - to which I could contribute my information.
- [X] I have read the FAQs inside the app (Menu -> About -> FAQs) and my problem isn't listed.
- [X] **I have taken the time to fill in all the required details. I understand that the bug report will be dismissed otherwise.**
- [X] This issue contains only one bug.
- [X] I have read and understood the [contribution guidelines](https://github.com/FossifyOrg/Calendar/blob/master/CONTRIBUTING.md).

### Affected app version

1.0.3

### Affected Android/Custom ROM version

Android 14

### Affected device model

Samsung Galaxy A52

### How did you install the app?

Google Play Store

### Steps to reproduce the bug

1. Create Widget Month View 4x4
2. Tap on any date on the widget
3. App opens with the day set to the current day

### Expected behavior

Tapping, e.g. on May 24th, should open the agenda view of the selected day. (see first screenshot)

### Actual behavior

Tapping, e.g. on May 24th, opens up the agenda view of the current day. (see second screenshot)

### Screenshots/Screen recordings

![Screenshot_20240515_095319_One UI Home-edit-20240515095434](https://github.com/FossifyOrg/Calendar/assets/137696320/28976fc0-7aec-4988-b491-3354cd69d1d8)
![Screenshot_20240515_095332_Calendar](https://github.com/FossifyOrg/Calendar/assets/137696320/60b793bc-a4a9-48e9-ae80-c2979d9a9ed5)

### Additional information

- Issue may be related, but not identical to #42

**Comment by Aga-C** [comment]
I see on your screen the toast message about updating birthdays. Do you also encounter this problem when the toast doesn't show?

**Comment by duerrhannes** [comment]
Indeed, when the toast message for adding contact birthdays does not show, the behaviour is correct and as expected.

(off topic: the toast message appears every time the app is opened when "add new birthdays automatically" is active)
